src/components/helpers/figure_drag_helper.ts

```diff
@@ -0,0 +1,77 @@
+import { MIN_FIG_SIZE } from "../../constants";
+import { Figure, FIGURE_BORDER_SHIFT, PixelPosition, SheetScrollInfo } from "../../types";
+
+export function dragFigureForMove(
+  initialMousePosition: PixelPosition,
+  currentMousePosition: PixelPosition,
+  initialFigure: Figure,
+  mainViewportPosition: PixelPosition,
+  scrollInfo: SheetScrollInfo
+): Figure {
+  const initialMouseX = initialMousePosition.x;
+  const mouseX = currentMousePosition.x;
+  const viewportX = mainViewportPosition.x;
+  const frozenPaneOffset = FIGURE_BORDER_SHIFT;
+
+  const initialMouseY = initialMousePosition.y;
+  const mouseY = currentMousePosition.y;
+  const viewportY = mainViewportPosition.y;
+
+  const deltaX = initialMouseX - mouseX;
+  let newX = initialFigure.x - deltaX;
+
+  // Freeze panes: always display the figure above the panes
+  if (viewportX > 0) {
```

Comment:
we could maybe factorize the X/Y computations to aleviate the functions, it would spare around 10 lines each time but would also put the complexity at one specific place, and not double it